Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/adapters/GameAdapter.kt

```diff
@@ -0,0 +1,174 @@
+import org.dolphinemu.dolphinemu.services.GameFileCacheManager
+import org.dolphinemu.dolphinemu.R
+import android.view.animation.AnimationUtils
+import org.dolphinemu.dolphinemu.activities.EmulationActivity
+import androidx.fragment.app.FragmentActivity
+import org.dolphinemu.dolphinemu.databinding.CardGameBinding
+import org.dolphinemu.dolphinemu.dialogs.GamePropertiesDialog
+import org.dolphinemu.dolphinemu.features.settings.model.BooleanSetting
+import java.util.ArrayList
+
+class GameAdapter(activity: Activity) : RecyclerView.Adapter<GameViewHolder>(),
+    View.OnClickListener, OnLongClickListener {
+    private var mGameFiles: List<GameFile>
+    private val mActivity: Activity
+
+    /**
+     * Initializes the adapter's observer, which watches for changes to the dataset. The adapter will
+     * display no data until swapDataSet is called.
+     */
+    init {
+        mGameFiles = ArrayList()
+        mActivity = activity
+    }
+
+    /**
+     * Called by the LayoutManager when it is necessary to create a new view.
+     *
+     * @param parent   The RecyclerView (I think?) the created view will be thrown into.
+     * @param viewType Not used here, but useful when more than one type of child will be used in the RecyclerView.
+     * @return The created ViewHolder with references to all the child view's members.
+     */
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): GameViewHolder {
+        val binding = CardGameBinding.inflate(LayoutInflater.from(parent.context))
+        binding.root.apply {
+            setOnClickListener(this@GameAdapter)
+            setOnLongClickListener(this@GameAdapter)
+        }
+
+        // Use that view to create a ViewHolder.
+        return GameViewHolder(binding)
+    }
+
+    /**
+     * Called by the LayoutManager when a new view is not necessary because we can recycle
+     * an existing one (for example, if a view just scrolled onto the screen from the bottom, we
+     * can use the view that just scrolled off the top instead of inflating a new one.)
+     *
+     * @param holder   A ViewHolder representing the view we're recycling.
+     * @param position The position of the 'new' view in the dataset.
+     */
+    override fun onBindViewHolder(holder: GameViewHolder, position: Int) {
+        val context = holder.itemView.context
+        val gameFile = mGameFiles[position]
+
+        holder.apply {
+            if (BooleanSetting.MAIN_SHOW_GAME_TITLES.booleanGlobal) {
+                binding.textGameTitle.text = gameFile.title
+                binding.textGameTitle.visibility = View.VISIBLE
+                binding.textGameTitleInner.visibility = View.GONE
+                binding.textGameCaption.visibility = View.VISIBLE
+            } else {
+                binding.textGameTitleInner.text = gameFile.title
+                binding.textGameTitle.visibility = View.GONE
+                binding.textGameCaption.visibility = View.GONE
+            }
+        }
```

Comment:
This wasn't here before. If you want to move this, please do it outside of the first commit.